src/dataflow/src/source/kafka.rs

```diff
@@ -49,6 +62,10 @@ pub struct KafkaSourceReader {
     last_offsets: HashMap<i32, i64>,
     /// Map from partition -> offset to start reading at
     start_offsets: HashMap<i32, i64>,
+    /// Timely worker logger for source events
+    logger: Option<Logger>,
```

Comment:
hmmm I think this is fine but maybe a better interface would be to add a function called `emit_logs` or something like that to the `SourceReader` trait which then `ConsistencyInfo` or something else that holds onto the logger can actually log.

Reply:
not something that needs to happen in this pr though!

Reply:
I guess another nice thing about having a separate function is that we can then amortize the logs and only emit them once per operator execution instead of once per incoming message

Reply:
Perhaps this might be something to explore if we find ourselves adding more source types that want to log "source specific" things?

Reply:
sure yeah! as long as this seems to pass our load tests without a regression I'm fine with it :D

Reply:
Cool! I'm running the avro upsert benchmark now. Previous testing didn't seem to show much of a delta with this logging enabled. Should have full results tomorrow.